components/homme/src/share/gllfvremap_util_mod.F90

```diff
@@ -685,27 +686,32 @@ function gfr_pgn_to_smoothed_topo(par, elem, output_nphys, intopofn, outtopopref
     if (write_latlon) nvar = 3
 
     allocate(gll_fields(np,np,nelemd,nvar), pg_fields(np*np,nelemd,nvar))
+    hybrid = hybrid_create(par, 0, 1)
+
 
     ! Read the unsmoothed physgrid topo data from cube_to_target's first
     ! run. Here, pg4 will give the best quality.
     fieldnames(1) = 'PHIS'
+    if (hybrid%masterthread) print *,'Attempting to read PG4 PHIS data for smoothing:'
     call read_physgrid_topo_file(intopofn, elem, par, fieldnames, intopo_nphys, pg_fields, stat)
-    if (stat /= 0) return
-
-    ! Map this topo field to GLL.
-    call gfr_init(par, elem, intopo_nphys)
-    call gfr_fv_phys_to_dyn_topo(par, elem, pg_fields(:,:,1))
-    do ie = 1,nelemd
-       call edgeVpack_nlyr(edge_g, elem(ie)%desc, elem(ie)%state%phis, 1, 0, 1)
-    end do
-    call bndry_exchangeV(par, edge_g)
-    do ie = 1,nelemd
-       call edgeVunpack_nlyr(edge_g, elem(ie)%desc, elem(ie)%state%phis, 1, 0, 1)
-    end do
-    call gfr_finish()
+    if (stat == 0) then
+       ! Map this topo field to GLL.
+       call gfr_init(par, elem, intopo_nphys)
+       call gfr_fv_phys_to_dyn_topo(par, elem, pg_fields(:,:,1))
+       do ie = 1,nelemd
+          call edgeVpack_nlyr(edge_g, elem(ie)%desc, elem(ie)%state%phis, 1, 0, 1)
+       end do
+       call bndry_exchangeV(par, edge_g)
+       do ie = 1,nelemd
+          call edgeVunpack_nlyr(edge_g, elem(ie)%desc, elem(ie)%state%phis, 1, 0, 1)
+       end do
+       call gfr_finish()
+    else
+       ! error trying to read pg4 input data. try GLL:
```

Comment:
This comment should have just `pg`, not `pg4`. pg4 input is not required; pg1,2,3 all work.